src/StoragelessSession/Session/Data.php

```diff
@@ -33,14 +33,49 @@ class Data implements \JsonSerializable
     private $metadata;
 
     /**
+     * @var bool
+     */
+    private $dataWasChanged = false;
+
+    /**
      * @todo ensure serializable data?
      */
     private function __construct(array $data, array $metadata)
     {
+        $this->verifyExpirationTime($data, $metadata);
         $this->data     = $data;
         $this->metadata = $metadata;
     }
 
+    /**
+     * @param array $data
+     * @param array $metadata
+     *
+     * @return bool
+     */
+    private function verifyExpirationTime(array &$data, array &$metadata)
+    {
+        if (isset($metadata['EXPIRE']) && ($_SERVER['REQUEST_TIME'] > $metadata['EXPIRE'])) {
```

Comment:
That breaks the statement in `README.md`:

>You can do this also in asynchronous contexts and long running processes, since no super-globals nor I/O are involved.